test/functional/aws_legacy_cluster_vpc.go

```diff
@@ -0,0 +1,284 @@
+	newErr := &networkConfigTestError{
+		subnetsError: []error{},
+	}
+
+	// filter subnets by integreatly cluster id tag
+	describeSubnets, err := session.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(clusterVPCId)},
+			},
+		},
+	})
+	if err != nil {
+		errMsg := fmt.Errorf("could not describe subnets: %v", err)
+		newErr.subnetsError = append(newErr.subnetsError, errMsg)
+		return nil, newErr
+	}
+
+	return describeSubnets.Subnets, newErr
+}
+
+func verifyLegacyRDSSubnetGroups(rdsSvc *rds.RDS, subnetGroupName, clusterVPCId string, vpcSubnets []*ec2.Subnet, azs []*string) error {
+	newErr := &networkConfigTestError{
+		rdsSubnetGroupsError: []error{},
+	}
+
+	// get rds subnet groups by subnet group name
+	describeGroups, err := rdsSvc.DescribeDBSubnetGroups(&rds.DescribeDBSubnetGroupsInput{
+		DBSubnetGroupName: aws.String(subnetGroupName),
+	})
+
+	if err != nil {
+		errMsg := fmt.Errorf("error describing rds subnet groups: %v", err)
+		newErr.rdsSubnetGroupsError = append(newErr.rdsSubnetGroupsError, errMsg)
+		return newErr
+	}
+
+	// expect 1 subnet group
+	subnetGroups := describeGroups.DBSubnetGroups
+	if len(subnetGroups) != 1 {
+		errMsg := fmt.Errorf("unexpected number of rds subnet groups: %d", len(subnetGroups))
+		newErr.rdsSubnetGroupsError = append(newErr.rdsSubnetGroupsError, errMsg)
+		return newErr
+	}
+
+	subnetGroup := subnetGroups[0]
+
+	// ensure the subnet group belongs to the cluster VPC
+	if *subnetGroup.VpcId != clusterVPCId {
+		errMsg := fmt.Errorf("rds subnet group %s does not belong to cluster VPC. got = %s, wanted = %s", *subnetGroup.DBSubnetGroupName, *subnetGroup.VpcId, clusterVPCId)
+		newErr.rdsSubnetGroupsError = append(newErr.rdsSubnetGroupsError, errMsg)
+	}
+
+	// ensure each subnet belongs to the vpc
+	for _, subnet := range subnetGroup.Subnets {
+		if !subnetExistsInVPCSubnetList(*subnet.SubnetIdentifier, vpcSubnets) {
+			errMsg := fmt.Errorf("rds subnet group %+v has a subnet that doesn't belong to the cluster VPC %s", subnetGroup, clusterVPCId)
+			newErr.rdsSubnetGroupsError = append(newErr.rdsSubnetGroupsError, errMsg)
+		}
+	}
+
+	// the list of availability zones that have been covered by the subnets
+	// within the DBSubnetGroup
+	coveredAZs := make([]*string, 0)
+	for _, subnet := range subnetGroup.Subnets {
+		if !contains(coveredAZs, subnet.SubnetAvailabilityZone.Name) {
+			coveredAZs = append(coveredAZs, subnet.SubnetAvailabilityZone.Name)
+		}
+	}
+
+	// check the lengths match, i.e. the subnets cover every AZ in the region
+	if len(coveredAZs) != len(azs) {
+		errMsg := fmt.Errorf("rds subnet group does not have a subnet group for each availability zone. Availability Zones: %+v Subnet Groups: %+v", subnetGroup.Subnets, azs)
+		newErr.rdsSubnetGroupsError = append(newErr.rdsSubnetGroupsError, errMsg)
+	}
+
+	return newErr
+}
```

Comment:
I'd like to get people's thoughts on the approach here.

Basically I'm verifying the following

* That the RDS subnet group belongs to the cluster VPC
* That each subnet in the group belongs is found in the list of subnets queried with the VPC id as a filter (this step is probably unnecessary but there's no harm in it)
* build a unique list of availability zones covered by the subnets and ensure it's same length as the actual list of AZs, i.e. the subnets cover each AZ.

The same thing is repeated for elasticache.

Is there anything else that needs to be checked?

Reply:
I think once we've addressed https://github.com/integr8ly/integreatly-operator/pull/946/files#r449061244 it's all good!